frappe/integrations/doctype/connected_app/connected_app.py

```diff
@@ -147,6 +148,35 @@ def get_active_token(self, user=None):
 
 		return token_cache
 
+	def get_backend_app_token(self):
+		"""Get an Access Token for the Cloud-Registered Service Principal"""
+		# There is no User assigned to the app, so we give it an empty string,
+		# otherwise it will assign the logged in user.
+		token_cache = self.get_token_cache("")
+		if token_cache is None:
+			token_cache = frappe.new_doc("Token Cache")
+			token_cache.connected_app = self.name
+		elif not token_cache.is_expired():
+			return token_cache
+
+		# Get a new Access token for the App
+		client = BackendApplicationClient(
+			client_id=self.client_id,
+			scope=self.get_scopes()
+		)
+		oauth_session = OAuth2Session(client=client)
+
+		token = oauth_session.fetch_token(
+			self.token_uri,
+			client_secret = self.get_password("client_secret")
+		)
+
+		token_cache.update_data(token)
+		token_cache.save(ignore_permissions=True)
+		frappe.db.commit()
```

Comment:
Did it not work without the explicit commit?

Ideally all POST requests would commit the transaction.

Reply:
Went through the flow now, some callers might be GET requests - so its fine.